framework/auth/core.py

```diff
@@ -843,6 +848,22 @@ def _get_unconfirmed_email_for_token(self, token):
 
         return verification['email']
 
+    def confirm_token(self, token):
+        """Return whether or not a confirmation token is valid for this user.
+        :rtype: bool
+        """
```

Comment:
Because this method returns a boolean, there's no way to differentiate between (a) token doesn't exist and (b) token is expired.

Instead of returning `False` what if you return different custom exceptions?